### PR TITLE
Added a Create User page for Frontend UI

### PR DESCRIPTION
**Pull request by c-lawrence**
References #18, #34, #50, #51

Implements a new Create Account page to the web application, linked to the login page. Page includes fields for required user info, and directs to the landing page when valid account details are entered. Needs to be linked to the backend Users database.

### CODE REVIEW

File: front-end/src/components/newAccountForm.tsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+    //All fields must be filled
+    if (firstNameField  === "" || lastNameField  === "" || emailField === "" || passwordField === "" || passwordField2 === "") {
+      setAlertReason("Enter your name, email address, and password.");
+      return;
+    }
+
+    //Check username/password requirements and security for text fields.
+    //Firstname and Lastname forms must use appropriate signage (alpha characters only)
+    if (!(firstNameField.match(checkSignage)) || !(lastNameField.match(checkSignage)) ) {
```
Comment (jacksontheel) [inline front-end/src/components/newAccountForm.tsx:86]: Maybe modify this to allow hyphens, and maybe apostrophes? We want Joseph Gordon-Levitt  to be able to use our site. 😂

File: front-end/src/components/newAccountForm.tsx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    }
+
+    //Passwords must match
+    if (passwordField !== passwordField2) {
+      setAlertReason("Your passwords must match.");
+      return;
+    }
+
+    //Passwords must be correct format (8-20 characters, 1 letter, 1 number, 1 special character, no spaces)
+    if (!(passwordField.match(checkPassSignage)) || !(passwordField2.match(checkPassSignage)) ) {
```
Comment (jacksontheel) [inline front-end/src/components/newAccountForm.tsx:104]: Don't need this or, we've already determined passwordField === passwordField2 by this point

File: front-end/src/components/newAccountForm.tsx (hunk 1 above)
Comment (c-lawrence) [reply front-end/src/components/newAccountForm.tsx:86]: Will do.

File: front-end/src/components/newAccountForm.tsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+  const checkSignage = /^[a-z]+$/i;
+
+  //Basic email signage.
+  const checkEmailSignage = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i
+
+  /* min eight characters,
+  max 20 characters, 
+  at least one letter, one number 
+  and one special character */
+  const checkPassSignage = /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&^])[A-Za-z\d@$!%*#?&^]{8,20}$/;
```
Comment (jacksontheel) [inline front-end/src/components/newAccountForm.tsx:69]: I think password should allow spaces.

File: front-end/src/components/newAccountForm.tsx [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+      <div>
+        <Button
+          onClick={handleSubmit}
+          style={styles.button}
+          variant="contained"
+        >
+          Create Account
+        </Button>
+      </div>
+      <Link href="/" underline="always">
```
Comment (jacksontheel) [inline front-end/src/components/newAccountForm.tsx:173]: Could we add a margin-up to the link? In this form and other one.

**Review by jacksontheel — CHANGES_REQUESTED** [review]
Awesome work, just a couple suggestions

File: front-end/src/components/newAccountForm.tsx (hunk 1 above)
Comment (c-lawrence) [reply front-end/src/components/newAccountForm.tsx:86]: Just made a commit for this.

File: front-end/src/components/newAccountForm.tsx (hunk 2 above)
Comment (c-lawrence) [reply front-end/src/components/newAccountForm.tsx]: Changed the check to "!(passwordField.match(checkPassSignage))"

File: front-end/src/components/newAccountForm.tsx (hunk 3 above)
Comment (c-lawrence) [reply front-end/src/components/newAccountForm.tsx:69]: Do you mean "_" ?

Comment (c-lawrence) [reply front-end/src/components/newAccountForm.tsx]: Went ahead and allowed "_" as a valid character in a password.

Comment (c-lawrence) [reply front-end/src/components/newAccountForm.tsx]: Changed the max password length to 50.

File: front-end/src/components/newAccountForm.tsx (hunk 4 above)
Comment (c-lawrence) [reply front-end/src/components/newAccountForm.tsx:173]: Could you elaborate what you mean? I'm not sure what "a margin-up" is supposed to be, sorry.

Comment (c-lawrence) [reply front-end/src/components/newAccountForm.tsx]: Resolved the issue by adding a new margin into the field.

**Review by jacksontheel — APPROVED** [review]
👍

**Pull request merged**